### PR TITLE
Add combo tracking

### PR DESCRIPTION
**Pull request by gustaveen**
Broke out the gtm tracking and added tracking for norwegian combo quotes

### CODE REVIEW

File: src/client/utils/tracking/gtm.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+export interface DataLayerObject {
+  event: string
+  offerData?: GTMOfferData
+}
+
+const pushToGTMDataLayer = (obj: DataLayerObject) => {
+  window.dataLayer = window.dataLayer || []
+  window.dataLayer.push(obj)
```
Comment (gustaveen) [inline src/client/utils/tracking/gtm.ts:26]: TS started to complain all of a sudden about `Property 'dataLayer' does not exist on type 'Window & typeof globalThis'` but I only moved the file into a folder? 🤷‍♂️ @palmenhq It is defined in gtm.d.ts

Comment (palmenhq) [reply src/client/utils/tracking/gtm.ts:26]: i'd just cast `window` to `any` in this particular case 🙈

Comment (gustaveen) [reply src/client/utils/tracking/gtm.ts]: yup let's just do that 😄

File: src/client/utils/tracking/gtm.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  castedWindow.dataLayer = castedWindow.dataLayer || []
+  castedWindow.dataLayer.push(obj)
+}
+
+export const trackOfferGTM = (
+  offerData: OfferData,
+  referralCodeUsed: boolean,
+) => {
+  try {
+    const contractType = () => {
```
Comment (palmenhq) [inline src/client/utils/tracking/gtm.ts:36]: Maybe we cold rename this to `getContractType` or `mapContractType` or something? Just to imply this variable is a funciton

**Review by palmenhq — COMMENTED** [review]
Sorry, two more thoughts 😅

File: src/client/pages/OfferNew/utils.ts [hunk 3]
```diff
@@ -108,9 +108,8 @@ export const isBundle = (offerData: OfferData): boolean =>
 export const isYouth = (offerData: OfferData): boolean =>
   offerData.quotes.every(
     (quote) =>
-      isNorwegianHomeContents(quote.quoteDetails) &&
-      isNorwegianTravel(quote.quoteDetails) &&
-      quote.quoteDetails.isYouth,
+      isNorwegianHomeContents(quote.quoteDetails) ||
+      (isNorwegianTravel(quote.quoteDetails) && quote.quoteDetails.isYouth),
```
[the remark this answers is not in the document]
Comment (gustaveen) [reply src/client/pages/OfferNew/utils.ts:112]: true, that order would make more sense. but the reason is `Property 'isYouth' does not exist on type 'SwedishApartmentQuoteDetails'` But it feels like there's a much simpler solution I can't think of right now haha

File: src/client/utils/tracking/gtm.ts (hunk 2 above)
Comment (gustaveen) [reply src/client/utils/tracking/gtm.ts:36]: indeed!

**Review by palmenhq — APPROVED** [review]

**Pull request merged**